U6944D5GU : <@U5U4MSHC1> <@U6829S5NZ> :taco:
U5U4MSHC1 : `Sorted` sorts the dictionary based on the keys in increasing order since you are passing the list of keys here, here it is in alphabetical order.
U6944D5GU : yup i got it :slightly_smiling_face:!
U5U4MSHC1 : Cool :thumbsup:
U653RFLBD : good luck. when google hires you, don't forget us little guys :stuck_out_tongue_winking_eye:
U6829S5NZ : <@U6944D5GU> to elaborate more on <@U5U4MSHC1> ‘s answer, the function `marks.keys()` returns a list of keys in the dictionary and `sorted()` returns the sorted list. Although why is there a conversion to list when it will return a list anyways?
U6944D5GU : <@U6829S5NZ> mmm good point..
U6944D5GU : idk either
U6829S5NZ : Good luck,
U6829S5NZ : interesting to note is that it doesn't effect your dictionary in any manner, it will return a list of sorted elements anyways. even if you pass a string to `sorted()` function but if you use the `list.sort()` function , it changes the list.
U2BS4M1RV : From <https://docs.python.org/3/library/functools.html>:```
from functools import partial
basetwo = partial(int, base=2)
basetwo.__doc__ = 'Convert base 2 string to an int.'
help(basetwo)

Help on partial object:
class partial(builtins.object)
....

basetwo.__doc__
'Convert base 2 string to an int.'
```

U2BS4M1RV : The example given would seem to imply the doc string would be read for help, however, help returns the doc string for partial.
U68L8RAG0 : <@U653RFLBD> <@U6829S5NZ>, we'll see.. I still have a lot of code challenges to get through. Either way, I'm learning how to be better with Python.. I really appreciate the help. :slightly_smiling_face:
U69CDE5LJ : Hi, Can someone please help me with installing ipython on mac?
U5NFCHDT4 : <@U69CDE5LJ> sure.   install anaconda which will have everything and won't mess with your system's python.  <https://www.continuum.io/downloads#macos>
U5NFCHDT4 : It has ipython and so many other libs which you will use eventually.
U5NFCHDT4 : ipython or jupyter notebook makes a list of inputs (In) and outputs (Out)  so you can reference previous inputs and outputs by doing stuff like:  print(Out[1])   #returns  [‘English', ‘Math', ‘Science']
U5NFCHDT4 : If you have long lines or large code blocks it can save some scrolling/typing.
U69CDE5LJ : ok thank you
U5CGPBF0U : So, I have an import error that I've been trying to solve for the past three days and still haven't been able to figure out. I'm building a REST API and I've put my source into its own module titled `broadway`. I'm implementing an app factory pattern. In `__main__.py` I have `from broadway import create_app, models, util`, and `create_app` is defined in `__init__.py`.
These are the entire contents of `__init__.py`:
```from flask import Flask
from . import config
from .external import init_app

# App factory function
def create_app(config_data):
    app = Flask('broadway')
    app.config.from_object(config.DevelopmentConfig)
    app.config['SQLALCHEMY_TRACK_MODIFICATIONS'] = False
    external.init_app(app)
    return app```
When running my application locally with `python -m broadway` the application returns the following error: `from broadway import create_app, models, util - ModuleNotFoundError: No module named 'broadway'` (edited)

Strangely commenting out `app.config.from_object(config.DevelopmentConfig)` in `__init__.py` fixes the error. This is not a real solution of course since then my app is missing vital configuration info. Also, in this example the parameter (`config.DevelopmentConfig`) is hard-coded as originally `config_data` was passed to it but I changed it to make sure the data being passed wasn't the actual issue.

I just don't get how the line in `__init__.py` relates to the import error at all.

Also, I put a bunch of `print()`s throughout my code to see what was going on. They _all_ printed, even the ones after the actual `create_app` function runs in `__main__.py`.

It was suggested that this was a problem with the Flask development server. After running my app with the Flask CLI it launched without errors. Then the problem was that all the endpoints of my API returned 404s, even though there were no errors in the actual Python reported (Flask's debug mode was on).

U5UKKT0EP : <@U14NMBU3W> What s the files/folder structure of your project?
U0LSCQQNR : <@U5CGPBF0U> you better put up a github link
U0LSCQQNR : for your project
U0LSCQQNR : you've been posting about this for what, three days now?
U5CGPBF0U : <@U0LSCQQNR> Yep, had another dev take a look too but no luck. If I remove all the irrelevant/proprietary stuff I can fit it all in a snippet though.
U23NW24GY : Hello guys, I was wondering if its possible to use several class to write to the same figure in `matplotlib`?Say for example I have the following classes
```Client calls Boat.draw 
Boat.draw calls Sailors.draw```

U68Q8495W : Hi Guys,
I am trying to achieve a functionality where I can open a `new command prompt` and run some batch commands. Then redirecting the output to a `logfile` and `console` on real time. So I am using `wtee.exe` (ex command: `dir 2&gt;&amp;1 | wtee.exe logFile.txt`).

Choosing `popen` over `os.system` because I need control over the new process (new command prompt)

As I wont be able to use pipe symbol in Popen. I did the below functionality

```newConsole = Popen(
      ['cmd','/V:ON /K dir 2&gt;&amp;1'],
      creationflags=CREATE_NEW_CONSOLE,stdout=PIPE,stderr=STDOUT
      )
outputLog = Popen(
      [wtee.exe,'logFile.txt'],
      stdin=newConsole.stdout,
      )```
I can see the log file is generated and also the output in command prompt (on main window and not in new console).

So please help to find out the way how can I print the output in new console opened using subprocess.

U69HXQZJ9 : Hello
U0LSCQQNR : <@U68Q8495W> <https://stackoverflow.com/questions/15899798/subprocess-popen-in-different-console>


in above conversation, code/s mentioned has issue/s?
	If Yes:
	1.Bad	2.Very bad
	If No:
	1.Good	2.Very good


How confident are you?
	1.Low
	2.Average
	3.High

Optional!
can you highlight place/word/sentence which lead to your decision
